File: src/com/nerdscorner/android/plugin/utils/ThreadUtils.java
package com.nerdscorner.android.plugin.utils;

public class ThreadUtils {
    public static void cancelThread(Thread thread) {
        if (thread != null) {
            thread.interrupt();
        }
    }

    public static void cancelThreads(Thread... threads) {
        if (threads == null || threads.length == 0) {
            return;
        }
        for (Thread thread : threads) {
            cancelThread(thread);
        }
    }
}

File: src/com/nerdscorner/android/plugin/utils/ThreadUtils.kt
package com.nerdscorner.android.plugin.utils

fun Any.cancelThreads(vararg threads: Thread?) {
    if (threads.isNullOrEmpty()) {
        return
    }
    for (thread in threads) {
        thread.cancel()
    }
}

fun Any.executeDelayed(delay: Long = 0, block: () -> Unit): Thread {
    val thread = Thread {
        if (delay > 0) {
            try {
                Thread.sleep(delay)
                block()
            } catch (e: InterruptedException) {
                // Nothing to do here
            }
        } else {
            block()
        }
    }
    thread.start()
    return thread
}

fun Thread?.cancel() {
    this?.interrupt()
}

fun Any.startThread(block: () -> Unit) = executeDelayed(block = block)
